src/utils/math.hpp

```diff
@@ -77,10 +77,9 @@ constexpr T modulo(T num, int mod, T min = 0)
  *  round (base_damage * bonus / divisor) to the closest integer,
  *  but up or down towards base_damage
  */
-constexpr int round_damage(int base_damage, int bonus, int divisor) {
+constexpr int round_damage(double base_damage, int bonus, int divisor) {
 	if (base_damage==0) return 0;
-	const int rounding = divisor / 2 - (bonus < divisor || divisor==1 ? 0 : 1);
-	return std::max<int>(1, (base_damage * bonus + rounding) / divisor);
+	return std::max<int>(1, std::round((base_damage * bonus) / divisor));
```

Comment:
This removes rounding towards base_damage.

Reply:
to use std::round successfully I have to remove this variable, and if I leave it as it is the treatment of 0.5 becomes too random (truncated or rounded)

Reply:
What cases are you talking about?

Reply:
During a test, a value of 3.5 was truncated to 3 instead of being rounded to 4 as it systematically happens with std::round, hence the need to use it instead of the current variable

Reply:
If the base damage was below 3.5 then that is the correct thing to do... The point of this function is to round up or down based on if the final value is higher or lower.

Reply:
if base_damage is 3.5 what i talk

Reply:
I'm going to assume that you mean `round_damage(3.5, 100, 100)`. In that case it's fairly arbitrary which way to go and we could also go with rounding up and thus adjust the formula to `rounding = divisor / 2 - (bonus <= divisor || divisor==1 ? 0 : 1)`.

Reply:
i will try that

Reply:
it's work